python-avd/pyavd/_schema/models/avd_model.py

```diff
@@ -164,39 +169,54 @@ def __bool__(self) -> bool:
             for field in self._fields
         )
 
-    def _as_dict(self, include_default_values: bool = False, strip_values: tuple = (None, [], {})) -> dict:
+    def _strip_empties(self) -> None:
+        """In-place update the instance to remove data matching the given strip_values."""
+        for field, field_info in self._fields.items():
+            if (value := self._get_defined_attr(field)) is Undefined or field == "_custom_data":
+                continue
+
+            if issubclass(field_info["type"], AvdBase) and isinstance(value, AvdBase):
+                value._strip_empties()
+                if not value:
+                    delattr(self, field)
+                continue
+
+            if value is None:
+                delattr(self, field)
+
+    def _as_dict(self, include_default_values: bool = False) -> dict:
         """
         Returns a dict with all the data from this model and any nested models.
 
         Filtered for nested None, {} and [] values.
         """
         as_dict = {}
-        for field, field_info in self._fields.items() or ():
-            if (value := self._get_defined_attr(field)) is Undefined:
+        for field, field_info in self._fields.items():
+            value = self._get_defined_attr(field)
+
+            if field == "_custom_data":
+                if isinstance(value, dict) and value:
```

Comment:
Save one isinstance in some cases.. :)
```suggestion
                if value and isinstance(value, dict):
```

Reply:
but costs a bool check which is not always "free" for these classes...

Reply:
can _custom_data ever be a class?

Reply:
not really. It was this or cast() to make type-checking work.

Reply:
hmm ok so isinstance is only present for the sake of type-checking?

Let's resolve this

Reply:
I will refactor all these to use cast so it is more clear.

Reply:
done